flink-rpc/flink-rpc-akka/src/test/java/org/apache/flink/runtime/rpc/akka/ContextClassLoadingSettingTest.java

```diff
@@ -107,49 +108,87 @@ void shutdown() throws InterruptedException, ExecutionException, TimeoutExceptio
     void testAkkaRpcService_ExecuteRunnableSetsFlinkContextClassLoader()
             throws ExecutionException, InterruptedException {
         final CompletableFuture<ClassLoader> contextClassLoader = new CompletableFuture<>();
-        akkaRpcService.execute(
-                () -> contextClassLoader.complete(Thread.currentThread().getContextClassLoader()));
+        akkaRpcService
+                .getScheduledExecutor()
+                .execute(
+                        () ->
+                                contextClassLoader.complete(
+                                        Thread.currentThread().getContextClassLoader()));
         assertIsFlinkClassLoader(contextClassLoader.get());
```

Comment:
```suggestion
        assertThat(contextClassLoader.get()).isSameAs(pretendFlinkClassLoader);
```
Same here: We only need to check on the Flink classloader